libs/NfcCoreLib/lib/LibNfc/phLibNfc_Hci.c

```diff
@@ -82,7 +84,6 @@ phLibNfc_Sequence_t gphLibNfc_HciInitSequenceNci2x[] = {
 
 phLibNfc_Sequence_t gphLibNfc_HciChildDevCommonInitSequenceNci1x[] =
 {
-    { &phLibNfc_NfceeModeSet, &phLibNfc_NfceeModeSetProc },
     { &phLibNfc_HciGetHostList, &phLibNfc_HciGetHostListProc },
     { &phLibNfc_DelayForSeNtf, &phLibNfc_DelayForSeNtfProc },
     { &phLibNfc_HciGetHostTypeList,&phLibNfc_HciGetHostTypeListProc },
```

Comment:
The Sequence is meant only NCI 1.0, NCI 1.0 Mode set is performed only on HCI network, not on individual NFCEE. Delay will ensure all the notifications are received.

Reply:
It sounds like something change in the NFCEE/HCI network activation sequence for NCI1.0 with this patch. From what I understand from your comment and changes, before that patch Mode Set was potentially done for the HCI network and each individual NFCEE (as per gphLibNfc_HciInitSequenceNci1x and gphLibNfc_HciChildDevCommonInitSequenceNci1x sequence) while this patch looks to do Mode Set only on HCI network.
Are you sure that NFC_CX_DRIVER_HCI_NETWORK_PER_NFCEE will not do what you are trying to achieve here ?
I am wondering if this is not introducing a regression on the currently supported way of activating NFCEE at least in NCI1.0.

Reply:
Mode set is done for NFCEE irrespective of NCI 1.0 or NCI 2.0 after NFCEE/HCI activation.
To be specific the changes are done for NFCEE  ETSI-12 compliant, the mode set on NFCEE ensures there is no failure during APDU Pipe Creation. 
I see no regression due to above changes for activation NFCEE in NCI 1.0.

Reply:
I believe I have spotted at least 2 major regressions with this current change but I will have some comments and try to argue...

The NCI1x NFCEEs activation process starts with the HCIInitSequenceNci1x. It is supposed to be triggered to enable the HCI network so that the NFCC can inform the Dh about the available NFCEE.
Once this is done and with your current changes:
The next sequence to be executed is HciChildDevCommonInitSequenceNci1x which will directly retrieve the HostList. I currently expect it to be empty as the only activated NFCEE should the HCI network NFCEE which won't be a HCI host and as a consequence not listed in the HostList.
I don't understand what are we waiting for with phLibNfc_DelayForSeNtf in that sequence as no pending activation are supposed to be ongoing. As a side note the HCI network NFCEE activation is supposed to be completed at this stage.
HostTypeList is then used to evaluate if any ETSI v12 *eSE* are available in the network. Assuming the previous statement is correct, I don't expect any SE in the HostTypeList as non of them were previously activated. At this stage we know:
- From HciInitSequenceNci1x if the NFCC was ETSI v12 compliant.
- Eventually that we may have ETSI v12 eSE enabled.

On HciChildDevCommonInitSequenceNci1x completion, phLibNfc_HciChildDevCommonInitComplete will possibly enable ETSIv12 eSE NFCEE only if the NFCC is ETSIv12 compliant. I believe there are 2 regressions at this stage:
- Only ETSIv12 eSE are going to be enabled. '03 00' as host type being only used for eSE, UICC that was able to be enabled won't be anymore as they will have a different host type '02 00'.
Devices in the field were able to do card emulation using the UICC. It won't be possible here as it is not showing as a ETSIv12 compliant SE.
- Only ETSIv12 compliant NFCC will be able to enable an SE. I believe devices in the field may use non ETSIv12 compliant NFCC and this will just prevent card emulation through any secure element type to happen.

Reply:
Point 1.
NFCEE ETSI v12 Compliancy check is made to enable the Wired mode.
     a.If NFCEE is ETSI v9 Compliant there is proprietary APDU Gate where NFCC will create APDU Pipe.
   b. If NFCEE is ETSI12 Compliant then as per ETSI 12 specifications its DH responsibility to create APDU pipe on APDU Gate 0x30.

There is no effect on card Emulation if NFCEE   ETSI v9 /ETSI v12 still it will get emulated. 
These changes are meant only for Wired Mode.

Point 2.
If the NFCC is ETSI v9 it will not be able to create a APDU pipe to ETSI v12 compliant NFCEE and wired mode will not work. Still in this case also there is no effect on card emulation through secure element.

Coming to check of 02 00 we can extend it when wired mode is applicable on UICC. Currently all UICCs in market are ETSI v9 compliant and do not support wired mode.

PhLibNfc_HostModeSet will ensure that there is no failure during APDU pipe creation.

I see no regression in either case.